models/issues/issue_list.go

```diff
@@ -599,3 +599,55 @@ func (issues IssueList) GetApprovalCounts(ctx context.Context) (map[int64][]*Rev
 
 	return approvalCountMap, nil
 }
+
+func (issues IssueList) BlockingDependenciesMap(ctx context.Context) (issueDepsMap map[int64][]*DependencyInfo, err error) {
+	var issueDeps []*DependencyInfo
+
+	err = db.GetEngine(ctx).
+		Table("issue").
+		Join("INNER", "repository", "repository.id = issue.repo_id").
+		Join("INNER", "issue_dependency", "issue_dependency.issue_id = issue.id").
+		Where(builder.In("issue_dependency.dependency_id", issues.getIssueIDs())).
```

Comment:
I don't know if its really such a performance killer
Because its already filter for repo and issue_id and then there shouldn't be such a large number of deps to be searched for
But if you want I can add two indecies